File: Proyecto/Calendario_App/src/main/java/org/example/calendario_app/dao/EtiquetaDAO.java
package org.example.calendario_app.dao;

import org.example.calendario_app.model.Etiqueta;
import java.util.List;

public interface EtiquetaDAO {
    List<Etiqueta> findAllByUsuarioId(int idUsuario);

    int save(Etiqueta etiqueta, int idUsuario, Integer idGrupo);

    void updateGroupId(int labelId, Integer groupId);

    void delete(int id);
}

File: Proyecto_Android/Calendario_Android_App/app/src/main/java/com/example/calendario_android_app/dao/EtiquetaDAO.kt
package com.example.calendario_android_app.dao

import com.example.calendario_android_app.model.Etiqueta

interface EtiquetaDAO {
    suspend fun insertEtiqueta(etiqueta: Etiqueta, idUsuario: Int, idGrupo: Int?): Boolean
    suspend fun getEtiquetasByUsuario(idUsuario: Int): List<Etiqueta>
    suspend fun updateEtiquetaGrupo(idEtiqueta: Int, idGrupo: Int): Boolean
    suspend fun getGrupoIdByEtiqueta(idEtiqueta: Int): Int?
}
